src/Tribe/Commerce/PayPal/Main.php

```diff
@@ -1559,4 +1602,110 @@ public function get_attendees_by_order( $order ) {
 
 		return empty( $attendees ) ? array() : array_map( 'get_post', $attendees );
 	}
+
+	/**
+	 * Whether the ticket is a PayPal one or not.
+	 *
+	 * @since TBD
+	 *
+	 * @param Tribe__Tickets__Ticket_Object $ticket
+	 *
+	 * @return bool
+	 */
+	public function is_paypal_ticket( Tribe__Tickets__Ticket_Object $ticket ) {
+		return $ticket->provider_class === __CLASS__;
+	}
+
+	/**
+	 * Returns a list of attendees grouped by order.
+	 *
+	 * @since TBD
+	 *
+	 * @param int   $post_id
+	 * @param array $ticket_ids An optional array of ticket IDs to limit the orders by.
+	 *
+	 * @return array An associative array in the format [ <order_number> => <order_details> ]
+	 */
+	public function get_orders_by_post_id( $post_id, array $ticket_ids = null ) {
+		$attendees = $this->get_attendees_by_id( $post_id );
+
+		if ( empty( $attendees ) ) {
+			return array();
+		}
+
+		$stati     = $this->get_order_stati();
+		$undefined = Tribe__Utils__Array::get( $stati, 'undefined', reset( $stati ) );
+		/** @var Tribe__Tickets__Commerce__PayPal__Orders__Sales $sales */
+		$sales = tribe( 'tickets.commerce.paypal.orders.sales' );
+
+		$orders = array();
+
+		foreach ( $attendees as $attendee ) {
+			if ( ! empty( $ticket_ids ) && ! in_array( $attendee['product_id'], $ticket_ids ) ) {
+				continue;
+			}
+
+			$order_number = get_post_meta( $attendee['attendee_id'], $this->order_key, true );
+
+			if ( ! isset( $orders[ $order_number ] ) ) {
+				$orders[ $order_number ] = array(
+					'url'             => $this->get_order_url( $order_number ),
+					'number'          => $order_number,
+					'status'          => $attendee['order_status'],
+					'status_label'    => Tribe__Utils__Array::get( $stati, $attendee['order_status'], $undefined ),
+					'purchaser_name'  => $attendee['purchaser_name'],
+					'purchaser_email' => $attendee['purchaser_email'],
+					'purchase_time'   => $attendee['purchase_time'],
+					'attendees'       => array( $attendee ),
+				);
+			} else {
+				$orders[ $order_number ]['attendees'][] = $attendee;
+			}
+		}
+
+		foreach ( $orders as &$order ) {
+			$order['line_total'] = $sales->get_revenue_for_attendees( $order['attendees'] );
+		}
+
+		return $orders;
+	}
+
+	/**
+	 * Returns the list of PayPal tickets order stati.
+	 *
+	 * @since TBD
+	 *
+	 * @return array An associative array in the [ <slug> => <label> ] format.
+	 */
+	public function get_order_stati() {
```

Comment:
should use `statuses` instead of `stati` -- idk what the LOE would be on implementing that change, but if it's small, i'd recommend making the change (no worries if not though)

Reply:
I'm sticking with a [WordPress example here](https://codex.wordpress.org/Function_Reference/get_post_stati); I see both used but since English plural is formally *statuses* I will change it.